py/server/deephaven/table.py

```diff
@@ -3555,3 +3560,101 @@ def update_by(self, ops: Union[UpdateByOperation, List[UpdateByOperation]],
+        """Creates a new MultiJoinTable. The join can be specified in terms of either tables or MultiJoinInputs.
+
+        Args:
+            input (Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]]): the input objects
+                specifying the tables and columns to include in the join.
+            on (Union[str, Sequence[str]], optional): the column(s) to match, can be a common name or an equality
+                expression that matches every input table, i.e. "col_a = col_b" to rename output column names.  When
+                using MultiJoinInput objects, this parameter is ignored.
+
+        Raises:
+            DHError
+        """
+        try:
+            if isinstance(input, Table) or (isinstance(input, Sequence) and all(isinstance(t, Table) for t in input)):
+                tables = to_sequence(input, wrapped=True)
+                with auto_locking_ctx(*tables):
+                    j_tables = to_sequence(input)
+                    self.j_multijointable = _JMultiJoinFactory.of(on, *j_tables)
+            elif isinstance(input, MultiJoinInput) or (isinstance(input, Sequence) and all(isinstance(ji, MultiJoinInput) for ji in input)):
+                wrapped_input = to_sequence(input, wrapped=True)
+                tables = [ji.table for ji in wrapped_input]
+                with auto_locking_ctx(*tables):
+                    input = to_sequence(input)
+                    self.j_multijointable = _JMultiJoinFactory.of(*input)
+            else:
+                raise DHError(message="input must be a Table, a sequence of Tables, a MultiJoinInput, or a sequence of MultiJoinInputs.")
+
+        except Exception as e:
+            raise DHError(e, "failed to build a MultiJoinTable object.") from e
+
+
+
+def multi_join(input: Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]],
+               on: Union[str, Sequence[str]] = None) -> MultiJoinTable:
+    """ The multi_join method creates a new table by performing a multi-table natural join on the input tables.  The result
+    consists of the set of distinct keys from the input tables natural joined to each input table. Input tables need not
+    have a matching row for each key, but they may not have multiple matching rows for a given key.
+
+    Args:
+        input (Union[Table, Sequence[Table], MultiJoinInput, Sequence[MultiJoinInput]]): the input objects specifying the
+            tables and columns to include in the join.
+        on (Union[str, Sequence[str]], optional): the column(s) to match, can be a common name or an equality expression
+            that matches every input table, i.e. "col_a = col_b" to rename output column names.  When using MultiJoinInput
+            objects, this parameter is ignored.
+
+    Returns:
+        MultiJoinTable: the result of the multi-table natural join operation. To access the underlying Table, use the
+            table() method.
+    """
+    return MultiJoinTable(input, on)
```

Comment:
General API question.  I don't know the answer, but I'm asking it....

Is there any reason the user would need to know about `MultiJoinInputTable`?  Right now, the query syntax looks like:
```
t = multi_join([t1,t2,t3]).table()
```
Any reason it shouldn't just be
```
t = multi_join([t1,t2,t3])
```

Reply:
Yes, but only because in the future we intend to allow addition (and potentially removal) of tables from this join. 
```
mt = multi_join([t1,t2,t3]);
mt_new = mt.add_table(t4) # returns Immutable MultiJoinTable
```